test/metabase/public_settings_test.clj

```diff
@@ -370,3 +371,29 @@
             (public-settings/show-metabase-links! true)))
 
         (is (= true (public-settings/show-metabase-links)))))))
+
+(def ^:private after+value->expected {[true true]   false
+                                      [true false]  false
+                                      [false true]  true
+                                      [false false] false})
+
+(deftest show-updated-permission-banner-modal-test
+  (doseq [instance-creation-time [(t/local-date-time 2020) (t/local-date-time 2022)]
+          fifty-migration-time [(t/local-date-time 2021) (t/local-date-time 2023)]
+          banner-setting-value [true false]
+          modal-setting-value [true false]]
```

Comment:
I'm personally in favor of `clojure.test/are` for this kind of enumeration. You can get something like
```
(are [instance-creation 50-migration banner-setting modal-setting] (test-fn ...)
  2020 2021 true true false
  2020 2021 true false false
  ...)
```
(or whatever the real values should be)
That way you can easily see the expected value for each set of inputs without tracing through the test logic 🤷 won't block on this but just an idea

Reply:
I actually thought that the map would be have less moving parts, since there are only 4 situations that make a difference, rather than writing out all 16 and making devs do the math to figure out what actually determines the output. The expected value calculations are listed above, in https://github.com/metabase/metabase/pull/45076/files#diff-d7af27f3905be1f9d80534697e89be5de2a5635586e3099ced9afb7def4da18eR375-R378.

Reply:
Maybe it would be easier to read if I make it into a map. I'll do that